app/code/Magento/ConfigurableProductGraphQl/Model/Variant/Collection.php

```diff
@@ -83,18 +83,18 @@ public function __construct(
     }
 
     /**
-     * Add parent Id to collection filter
+     * Add parent to collection filter
      *
-     * @param int $id
+     * @param Product $product
      * @return void
      */
-    public function addParentId(int $id) : void
+    public function addParentId(Product $product) : void
     {
-        if (!in_array($id, $this->parentIds) && !empty($this->childrenMap)) {
+        if (!in_array($product, $this->parentProduct) && !empty($this->childrenMap)) {
```

Comment:
You should not use objects for `in_array` but you can use structure like key is product id, value is object